src/com/palmergames/bukkit/towny/command/TownCommand.java

```diff
@@ -4150,6 +4172,66 @@ public static void parseTownTrustCommand(CommandSender sender, String[] args, @N
 		town.save();
 	}
 
+	public static void parseTownTrustTownCommand(CommandSender sender, String[] args, @Nullable Town town) throws TownyException {
+
+		if (args.length < 1
+			|| args.length < 2 && (args[0].equalsIgnoreCase("add") || args[0].equalsIgnoreCase("remove"))
+			|| args.length == 1 && !args[0].equalsIgnoreCase("list")) {
+			HelpMenu.TOWN_TRUSTTOWN_HELP.send(sender);
+			return;
+		}
+		
+		if (town == null && sender instanceof Player player) {
+			town = getTownFromPlayerOrThrow(player);
+		}
+		if (args[0].equalsIgnoreCase("list")) {
+			TownyMessaging.sendMessage(sender, TownyFormatter.getFormattedTownyObjects(Translatable.of("status_trustedlist").forLocale(sender), new ArrayList<>(town.getTrustedTowns())));
+			return;
+		}
+
+		checkPermOrThrow(sender, PermissionNodes.TOWNY_COMMAND_TOWN_TRUSTTOWN.getNode());
+
+		Town trustTown = getTownOrThrow(args[1]);
+
+		if (args[0].equalsIgnoreCase("add")) {
+			if (town.hasTrustedTown(trustTown)) {
+				TownyMessaging.sendErrorMsg(sender, Translatable.of("msg_already_trusted", trustTown.getName(), Translatable.of("town_sing")));
+				return;
+			}
+			else if (town == trustTown) {
+				TownyMessaging.sendErrorMsg(sender, Translatable.of("msg_already_trusted", trustTown.getName(), Translatable.of("town_sing")));
+				return;
+			}
+			@Nullable Town finalTown = town;
+			Confirmation.runOnAccept(()-> {
+					for (Resident res : trustTown.getResidents()) {
+						plugin.deleteCache(res);
+					}
+
+					TownyMessaging.sendMsg(sender, Translatable.of("msg_trusted_added", trustTown.getName(), Translatable.of("town_sing")));
+					finalTown.addTrustedTown(trustTown);
+				})
+				.setTitle(Translatable.of("confirmation_msg_trusttown_consequences"))
+				.sendTo(sender);
+		} else if (args[0].equalsIgnoreCase("remove")) {

```

Comment:
```suggestion
		} else if ("remove".equalsIgnoreCase(args[0])) {
```
Literal "remove" is argument of 'equalsIgnoreCase()', instead of its qualifier